cmake/ConfigFileSetting.cmake

```diff
@@ -239,17 +239,15 @@ endif()
 # Copy cmake specific template files
 # =============================================================================
 # nrnconf.h.in and nmodlconf.h.in were originally generated from config.h.in generated by
-# autoheader. We use repository copy cmake_nrnconf.h.in to create these files.
-nrn_copy_file_if_different("${PROJECT_SOURCE_DIR}/cmake_nrnconf.h.in"
-                           "${PROJECT_SOURCE_DIR}/nrnconf.h.in")
-nrn_copy_file_if_different("${PROJECT_SOURCE_DIR}/cmake_nrnconf.h.in"
-                           "${PROJECT_SOURCE_DIR}/nmodlconf.h.in")
+# autoheader. We use repository copy cmake_nrnconf.h.in to create
+# nrnconf.h and nmodlconf.h directly in the PROJECT_BINARY_DIR from
+# the nrn_configure_file4 macro.
 
 # =============================================================================
 # Generate file from file.in template
 # =============================================================================
-nrn_configure_file(nrnconf.h .)
-nrn_configure_file(nmodlconf.h .)
+nrn_configure_file4(nrnconf.h . cmake_nrnconf.h .)
+nrn_configure_file4(nmodlconf.h . cmake_nrnconf.h .)
```

Comment:
Just a `nitpick` : can we rename this macro? something like `nrn_configure_file_rename`? or `nrn_configure_file_with_rename`?

Reply:
renamed to nrn_configure_dest_src.  This is a regrettably unusual macro with respect to args.
```nrn_configure_dest_src(bfile bdir sfile sdir)```
The first two define the destination ```PROJECT_BINARY_DIR/bdir/bfile```
and the last two define the source ```PROJECT_SOURCE_DIR/sdir/sfile.in```
A temporary intermediate file is created called
```PROJECT_BINARY_DIR/bdir/_cmake_tmp_bfile.in``` which is transformed via
configure_file into another intermediate for comparison with bfile with that intermediate being
called ```PROJECT_BINARY_DIR/bdir/_cmake_tmp_bfile
One can see that this latest incarnation of the macro does not really require the sfile sdir
pair as there is no need to have the sfile part of the filename in isolation as there was for
bfile